libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/MapboxCopilot.kt

```diff
@@ -0,0 +1,92 @@
+package com.mapbox.navigation.copilot
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.copilot.internal.PushStatusObserver
+import com.mapbox.navigation.core.DeveloperMetadataObserver
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * Mapbox Copilot is a Navigation SDK component that collects detailed trace files of navigation sessions together with search analytics data.
+ *
+ * Mapbox Copilot collects information that helps customers in two different ways:
+ * - It makes the feedback provided by drivers more actionable.
+ * - It provides Mapbox teams with the necessary information to investigate and resolve driver-reported issues and, consequently, helps expedite SDK updates that improve the driver experience for the entire developer community.
+ *
+ * It's a [MapboxNavigationObserver] so it's tied to [MapboxNavigation] lifecycle automatically.
+ * It's highly recommended to keep track of the [DeveloperMetadata.copilotSessionId] (see [DeveloperMetadataObserver]) so that Mapbox teams can easily act on specific end-users feedback as Mapbox teams can easily find the respective traces and troubleshoot issues faster.
```

Comment:
```suggestion
 * We recommended tracking the [DeveloperMetadata.copilotSessionId] (see [DeveloperMetadataObserver]) so that Mapbox teams can better act on specific end-user feedback. This ID helsp Mapbox teams find the respective traces and troubleshoot issues faster.
```